aws/rust-runtime/aws-inlineable/src/glacier_interceptors.rs

```diff
@@ -0,0 +1,454 @@
+            .modify_before_serialization(&mut context, &mut cfg)
+            .expect("success");
+        assert_eq!(
+            DEFAULT_ACCOUNT_ID,
+            context
+                .input()
+                .downcast_ref::<SomeInput>()
+                .unwrap()
+                .account_id
+                .as_ref()
+                .expect("it is set now")
+        );
+    }
+}
+
+#[cfg(test)]
+mod api_version_tests {
+    use super::*;
+    use aws_smithy_runtime_api::type_erasure::TypedBox;
+
+    #[test]
+    fn api_version_interceptor() {
+        let mut cfg = ConfigBag::base();
+        let context = InterceptorContext::<()>::new(TypedBox::new("dontcare").erase());
+        let mut context = context.into_serialization_phase();
+        let _ = context.take_input();
+        context.set_request(http::Request::builder().body(SdkBody::empty()).unwrap());
+        let mut context = context.into_before_transmit_phase();
+
+        let interceptor = GlacierApiVersionInterceptor::new("some-version");
+        interceptor
+            .modify_before_signing(&mut context, &mut cfg)
+            .expect("success");
+
+        assert_eq!(
+            "some-version",
+            context
+                .request()
+                .headers()
+                .get(API_VERSION_HEADER)
+                .expect("header set")
+        );
+    }
+}
+
+#[cfg(test)]
+mod precomputed_signable_body_tests {
+    use super::*;
+    use aws_runtime::auth::sigv4::SigningOptions;
+    use aws_smithy_runtime_api::type_erasure::TypedBox;
+
+    #[test]
+    fn precomputed_signable_body_interceptor() {
+        let mut cfg = ConfigBag::base();
+
+        let mut signing_options = SigningOptions::default();
+        signing_options.payload_override = None;
+        cfg.put(SigV4OperationSigningConfig {
+            region: "some-region".into(),
+            service: "some-service".into(),
+            signing_options,
+        });
+
+        let context = InterceptorContext::<()>::new(TypedBox::new("dontcare").erase());
+        let mut context = context.into_serialization_phase();
+        let _ = context.take_input();
+        context.set_request(
+            http::Request::builder()
+                .header(X_AMZ_CONTENT_SHA256, "content-sha256")
+                .body(SdkBody::empty())
+                .unwrap(),
+        );
+        let mut context = context.into_before_transmit_phase();
+
+        let interceptor = GlacierSetPrecomputedSignableBodyInterceptor::default();
+        interceptor
+            .modify_before_signing(&mut context, &mut cfg)
+            .expect("success");
+
+        assert_eq!(
+            &SignableBody::Precomputed("content-sha256".into()),
+            cfg.get::<SigV4OperationSigningConfig>()
+                .unwrap()
+                .signing_options
+                .payload_override
+                .as_ref()
+                .unwrap()
+        );
+    }
+}
+
+#[cfg(test)]
+mod treehash_checksum_tests {
```

Comment:
Is this the sort of thing where generating test cases with prop testing is useful?

Reply:
I don't think so in this case.